classes/submission/SubmissionAgencyVocab.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+/**
+ * @file classes/submission/SubmissionAgencyVocab.php
+ *
+ * Copyright (c) 2024 Simon Fraser University
+ * Copyright (c) 2024 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class SubmissionAgencyVocab
+ *
+ * @brief Operations for retrieving and modifying a submission's assigned agencies
+ */
+
+namespace PKP\submission;
+
+use APP\facades\Repo;
+use Illuminate\Database\Eloquent\Builder;
+use PKP\controlledVocab\ControlledVocab;
+use PKP\core\PKPApplication;
+
+
+class SubmissionAgencyVocab extends ControlledVocab
+{
+    public const CONTROLLED_VOCAB_SUBMISSION_AGENCY = 'submissionAgency';
+
+    /**
+     * Get the list of localized additional fields to store.
+     */
+    public function scopeGetLocaleFieldNames(Builder $query): array
+    {
+        return ['submissionAgency'];
+    }
+
+    /**
+     * Get agencies for a specified submission ID.
+     *
+     * @param int $assocType DO NOT USE: For <3.1 to 3.x migration pkp/pkp-lib#6213
+     * 
+     */
+    public function scopeGetAgencies(
```

Comment:
I feel like using the scope methods for these in these extended model classes does not make sense . It does not utilise the scope's methods capabilities and here we had to pass the `$builder` which has no use . Perhaps a simple static method is good enough .

Reply:
Just looking at the `DO NOT USE` note -- is this code that got left over after the 3.0.x to 3.1.x upgrade got removed? (Also more below.)